src/backend/joanie/payment/models.py

```diff
@@ -352,7 +352,22 @@ def get_card_for_owner(self, pk, owner_id):
         Retrieve a credit card for a given owner. If no such card exists, a
         CreditCard.DoesNotExist is raised.
         """
-        return self.get(pk=pk, owner_id=owner_id)
+        payment_provider = get_payment_backend()
+
+        return self.get(
+            pk=pk, owner_id=owner_id, payment_provider=payment_provider.name
+        )
+
+    def get_cards_for_owner(self, username):
+        """
+        Retrieve all the credit cards for a given owner that are tokenized by the
+        active payment provider
+        """
+        payment_provider = get_payment_backend()
+
+        return self.filter(
+            owner__username=username, payment_provider=payment_provider.name
+        )
```

Comment:
Ok, I did not catch the fact that signature of those methods are not equal. I think it's weird. Could we use a username for both ?

Reply:
I think it's a good idea to use username for both.